Sources/NIOHTTP1/HTTPTypes.swift

```diff
@@ -197,8 +286,18 @@ public struct HTTPHeaders: CustomStringConvertible {
     //      recommended.
     /// - Parameter value: The header field value to add for the given name.
     public mutating func add(name: String, value: String) {
-        let keyLower = name.lowercased()
-        storage[keyLower] = (storage[keyLower] ?? [])  + [(name, value)]
+        let nameStart = self.buffer.writerIndex
+
+        for c in name.utf8 {
```

Comment:
@weissi @Lukasa is there a better way ?

Reply:
I mean, you can use `lazy.map { $0.isASCII }.reduce(true) { $0 && $1 }`, but sadly that's only shorter, not necessarily faster.

Reply:
Yeah... that said we can add an add method that takes StaticString for name and check isASCII directly on it to provide a fast path

Reply:
@normanmaurer `name.utf8.contains(where: Character.isASCII)`